app/assets/javascripts/transitive_dependencies.js

```diff
@@ -0,0 +1,71 @@
+$(document).on('click', '.caret', function () {
+  var current = $(this)
+  var gem_id = $(this).attr('data-gem_id');
+  var version_id = $(this).attr('data-version');
+  $.ajax({
+    type: "get",
+    url: "/gems/"+gem_id+"/versions/"+version_id+"/dependencies.json",
+    success: function(resp) {
+      renderDependencies(resp, gem_id, current)
+    },
+    error: function() {
+      var error_message = "<ul class='deps_item--error'>Request failed. please reload the page and try again</ul>"
+      current.parent().next().next().html(error_message)
+    }
+  });
+})
+
+function renderDependencies(resp, gem_id, current) {
+  scope_display(current, gem_id, resp.run_deps, "runtime")
+  scope_display(current, gem_id, resp.dev_deps, "development")
+  arrow_toggler(current)
+}
+
+function arrow_toggler(current) {
+  var toggler = "<span class='deps_expanded deps_expanded-down'></span>";
+
+  current.parent().click(function() {
+    var runtime_div = $(this).parent().find('div').first()
+    runtime_div.toggleClass('deps_toggle');
+    runtime_div.next().toggleClass('deps_toggle');
+    $(this).find('span').first().toggleClass('deps_expanded-down');
+  });
+  current.parent().html(toggler);
+}
+
+function scope_display(current, gem_id, deps,scope) {
+  if (deps.length != 0){
+    var new_gems = current.parent().next().next()
+    if (scope == "development") { new_gems = new_gems.next() }
```

Comment:
Remove top div in following:
```
<div><div class="deps_scope"></div></div>
```
Use `deps_scope` selector with `closest` or `siblings` function instead of `current.parent().next().next()`. You can do away with `scope == "development"` check (and remove this function all together) if you use different selector for runtime and development scope/div.

Reply:
The reason I have a div before that is because the <span class='scope scope--expanded'>(Runtime/development) :</span> comes before the deps_scope div and inside the top div.

![Screenshot from 2019-08-25 11-00-54](https://user-images.githubusercontent.com/16337625/63645981-ace74380-c727-11e9-9363-d8305d7cd73c.png)

Reply:
I need that div for toggling scope and also the arrow